code/modules/chemistry/tools/dispensers.dm

```diff
@@ -557,6 +557,38 @@
 			return ..()
 
 
+/obj/reagent_dispensers/still/shittystill
+	name = "shitty still"
+	desc = "Looks like it some sort of makeshift distillery, made out of salvaged parts."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "shittystillFIN"
+	amount_per_transfer_from_this = 20
+	var/list/random_reagents_list = list("barium","flourine","lithium","magnesium","mercury",\
+"oxygen","phosphorus","plasma","potassium","radium","silver","sugar","hydrogen","water",\
+"carbon","chlorine","sulfur","calcium","iron","ethanol")
+
+	brew(var/obj/item/W as obj, mob/user as mob)
+		if (!W || !(istype(W,/obj/item/reagent_containers/food) || istype(W, /obj/item/plant)))
```

Comment:
checking if W is null, or if W is a plant here is not needed. A null / plant W will already fail the istype check.